db/migrate/20170302134438_fix_only_superseded_editions.rb

```diff
@@ -1,16 +1,20 @@
 class FixOnlySupersededEditions < ActiveRecord::Migration[5.0]
   def up
-    Document
+    docs = Document
       .joins(:editions)
       .where(%q(
         NOT EXISTS (
           SELECT id FROM editions AS e
           WHERE e.state IN ('draft', 'published', 'unpublished')
             AND e.document_id = documents.id
         )
-      )).find_each do |document|
-        document.editions.order(updated_at: :desc).first
-          .update_column(:state, "unpublished")
-      end
+      ))
+
+    docs.find_each do |document|
+      document.editions.order(updated_at: :desc).first
+        .update_column(:state, "unpublished")
```

Comment:
Hmm, yes, I think you're right. We don't know anything about what should be in the `Unpublishing` instance (`type`, `explanation`, `unpublished_at`, etc) though. Do you think a better fix would be to make the system cope with unpublished editions without an `Unpublishing` or just pick a suitable `type` and leave it at that.